server/middleware/token.js

```diff
@@ -32,6 +32,9 @@ function rewriteUserLiteral(req, currentUserLiteral) {
       debug('req.url has been rewritten from %s to %s', urlBeforeRewrite,
         req.url);
     }
+  } else if (!req.accessToken && currentUserLiteral &&
+      new RegExp(currentUserLiteral).test(req.url)) {
+    req.res.status(401).end();
```

Comment:
This will by-pass the application error handler, which will result in an inconsistent error response.

You should create a new `Error` object and pass it to `next` function which was provided to `token` middleware handler.